sdk/core/core-client-paging-rest/src/paginate.ts

```diff
@@ -42,67 +42,41 @@ export interface PaginateOptions {
  * @param options - Options to use custom property names for pagination
  * @returns - return a PagedAsyncIterableIterator that can be used to iterate the elements
  */
-export function paginateResponse<TReturn>(
+export function paginateResponse<TElement>(
   client: Client,
   initialResponse: HttpResponse,
   options: PaginateOptions = {}
-): PagedAsyncIterableIterator<TReturn, TReturn[]> {
-  const iter = listAll<TReturn>(client, initialResponse, options);
-  return {
-    next() {
-      return iter.next();
-    },
-    [Symbol.asyncIterator]() {
-      return this;
-    },
-    byPage: () => {
-      return listPage<TReturn>(client, initialResponse, options);
+): PagedAsyncIterableIterator<TElement> {
+  let firstRun = true;
+  const pagedResult: PagedResult<TElement[]> = {
+    firstPageLink: "",
+    async getPage(pageLink: string) {
+      let result: HttpResponse | undefined = undefined;
+      if (firstRun) {
+        result = initialResponse;
+        firstRun = false;
+      } else {
+        result = await client.pathUnchecked(pageLink).get();
+      }
```

Comment:
style nit: could be a little more terse here

```suggestion
      const result = firstRun ? initialResponse : await client.pathUnchecked(pageLink).get();
      firstRun = false;
```

Reply:
Done.